### ISSUE TITLE
Parsing error: Unexpected token, expected }

### ISSUE THREAD

**Issue by JustinTBrown** [eslint/eslint#10628]
**Tell us about your environment**

* **ESLint Version:** 4.10.0
* **Node Version:** 8.10.0
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": [
    "react-app",
    "plugin:flowtype/recommended",
    "plugin:jest/recommended",
    "prettier"
  ],
  "parser": "babel-eslint",
  "plugins": ["flowtype", "jest", "prettier"],
  "env": {
    "browser": true
  },
  "overrides": [
    {
      "files": ["src/**/*.spec.js"],
      "env": {
        "jest": true
      }
    }
  ],
  "rules": {
    "comma-dangle": ["error", "never"],
    "no-underscore-dangle": ["error", { "allowAfterThis": true }],
    "max-len": ["error", { "ignoreTrailingComments": true, "code": 100 }],
    "react/jsx-filename-extension": 0,
    "react/sort-comp": [
      "error",
      {
        "order": [
          "type-annotations",
          "state",
          "defaultProps",
          "static-methods",
          "lifecycle",
          "everything-else",
          "render"
        ]
      }
    ],
    "react/require-default-props": "off",
    "react/jsx-wrap-multilines": "off",
    "react/jsx-indent": "off",
    "react/jsx-indent-props": "off",
    "react/jsx-closing-bracket-location": "off",
    "class-methods-use-this": "off",
    "no-global-assign": 2,
    "no-param-reassign": "off",
    "no-plusplus": "off",
    "no-console": "error",
    "prettier/prettier": "error",
    "camelcase": [1, { "properties": "always" }],
    "no-unused-vars": [2, { "args": "after-used", "ignoreRestSiblings": true }]
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import React from "react";

const ChatWidget = () => {
  return (
    <script>
      window.fcWidget.init({
        token: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
        host: "https://wchat.freshchat.com"
      });
    </script>
  );
};

export default ChatWidget;
```

```bash
yarn lint:js
```
which runs `eslint src --max-warnings=0`

**What did you expect to happen?**
I expected eslint to not flag anything.

**What actually happened? Please include the actual, raw output from ESLint.**

```bash
yarn run v1.7.0
$ eslint src --max-warnings=0

/Users/justinbrown/Sync/code/iu-react/src/components/ChatWidget/index.js
  7:14  error  Parsing error: Unexpected token, expected }

   5 |     <script>
   6 |       window.fcWidget.init({
>  7 |         token: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
     |              ^
   8 |         host: "https://wchat.freshchat.com"
   9 |       });
  10 |     </script>

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
```

**Notes**
I've looked through a number of Stack posts as well as issues posted on this repo. I've tried setting ecmaVersion explicitly as well as other suggestions as proposed in posts like: https://stackoverflow.com/questions/37918769/eslint-parsing-error-unexpected-token and https://github.com/eslint/eslint/issues/10369

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue.

This is happening because your syntax is invalid. It looks like you're trying to use an inline `<script>` tag as a JSX element, but `<script>` tags in JSX don't work quite the same way that they do in HTML (for example, still have to follow JSX syntax, so they can't contain arbitrary JavaScript code).

As an immediate fix for your problem to get your code to parse, if you just want that code to run whenever your component is rendered, you could omit the `<script>` tag and simply do something like this:

```js
import React from "react";

const ChatWidget = () => {
  window.fcWidget.init({
    token: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
    host: "https://wchat.freshchat.com"
  });
  return <div> /* put any other UI to render here */ </div>;
};

export default ChatWidget;
```

As a more long-term fix, it generally isn't good practice for React components to have side-effects when rendering. Instead, you could initialize the widget in a `componentDidMount` method, with something like this:

```js
import React from "react";

class ChatWidget extends React.Component {
  componentDidMount() {
    window.fcWidget.init({
      token: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
      host: "https://wchat.freshchat.com"
    });
  }
  render() {
    return <div> /* put any other UI to render here */ </div>;
  }
}

export default ChatWidget;
```

**Comment by platinumazure** [comment]
Closing as the question appears to have been answered, but please leave a comment if any follow-up is needed and we can consider reopening.
